syntactic/src/test/scala/syntactic_checker/blacklist/BlacklistCheckerTests.scala

```diff
@@ -0,0 +1,90 @@
+package syntactic_checker.blacklist
+
+import org.junit.Assert._
+import org.junit.Test
+import syntactic_checker.CheckResult
+
+import scala.meta.dialects
+
+class BlacklistCheckerTests {
+
+  // TODO test requiring to load a file
+  // TODO cleaner tests
+
+  @Test
+  def nullInstanceOfCheckerTest(): Unit = {
+    val checker = BlacklistChecker(BlacklistRules.NoNull, BlacklistRules.NoCast)
+    val checkRes = checker.checkCodeString(dialects.Scala213, sourceCode)
+    checkRes match {
+      case CheckResult.ParsingError(e) => throw e
+      case CheckResult.Valid => fail("should not be valid")
+      case CheckResult.Invalid(detectedViolations) =>
+        val expected = List(
+          (11, 48),
+          (12, 51),
+          (22, 18),
+          (23, 20)
+        )
+        assertEquals(expected, detectedViolations.map(violation => (violation.startLine, violation.startColumn)))
+    }
+  }
+
+  @Test
+  def varWhileCheckerTest(): Unit = {
+    val checker = BlacklistChecker(BlacklistRules.NoVar, BlacklistRules.NoWhile)
+    val checkRes = checker.checkCodeString(dialects.Scala213, sourceCode)
+    checkRes match {
+      case CheckResult.ParsingError(e) => throw e
+      case CheckResult.Valid => fail("should not be valid")
+      case CheckResult.Invalid(detectedViolations) =>
+        val expected = List(
+          (2, 2),
+          (3, 2),
+          (16, 6),
+          (17, 6),
+          (20, 8),
+          (26, 6)
+        )
```

Comment:
Do you or (Chifan) plan to migrate these tests to the new infrastructure as a second step?

Reply:
Yes, having tests in this folder is only temporary, until the testing infrastructure is ready.